### PR TITLE
Read personnummer from Ladok API instead of "UNIX LDAP"

### PR DESCRIPTION
**Pull request by exacs**
This PR (which was a spin-off from migration to Stratus) is about reading "personnummer" information from Ladok API instead of UNIX LDAP.

It also includes other cleanups and upgrades (like bumping Node.js from 14 to 20)

### CODE REVIEW

File: server/export.js [hunk 1]
```diff
@@ -90,35 +90,32 @@ async function getAssignmentIdsAndHeaders({ canvasApi, canvasCourseId }) {
...
     row = {
       kthid: student.sis_user_id,
-      givenName: ugUser.givenName,
-      surname: ugUser.sn,
-      personnummer:
-        personnummer &&
-        (personnummer.length === 12 ? personnummer.slice(2) : personnummer),
+      givenName: ladokStudent?.givenName,
+      surname: ladokStudent?.surname,
+      personnummer: ladokStudent?.personnummer,
```
Comment (me97esn) [inline server/export.js:104]: This is a little different then before. If the student doesn't exist in Ladok, the values will be null.
Before, if the user didn't exist in ldap, an error was thrown.
This is an improvement I guess?

File: server/export.js [hunk 2]
```diff
@@ -90,35 +90,32 @@ async function getAssignmentIdsAndHeaders({ canvasApi, canvasCourseId }) {
...
     };
   } catch (err) {
     log.error(
-      `An error occured while trying to find user ${student.sis_user_id} in ldap`,
+      `An error occured while trying to find user ${student.sis_user_id} in Ladok`,
```
Comment (me97esn) [inline server/export.js:108]: Will this happen? If so: if an error is thrown when no user is found, when is ladokStudent on line 98 falsy? Do we need the null checks on line 102-104 or are they only adding unneeded complexity?

File: server/export.js [hunk 3]
```diff
@@ -347,7 +344,10 @@ async function exportResults3(req, res) {
     // Start writing response as soon as possible
     res.set({
       "content-type": "text/csv; charset=utf-8",
+      "Transfer-Encoding": "chunked",
       location: "http://www.kth.se",
+      "X-random-header": "value_value",
```
Comment (me97esn) [inline server/export.js:349]: This is left by mistake I assume?

File: server/submissions.js [hunk 4]
```diff
@@ -24,6 +24,7 @@ function extractUniqueStudentsAndSectionNamesFromSections(sections) {
               section_names: sectionName,
               submissions: [], // Add this to preserve the order of the properties, to make deepEqual possible
```
Comment (me97esn) [inline server/submissions.js:25]: Not part of this PR, and probably done by me back in the days, but damn this is ugly!

File: server/server.js [hunk 5]
```diff
@@ -55,7 +54,6 @@ server.use((req, res, next) => {
  */
 
 server.use(prefix, systemroutes);
-server.use(prefix + "/v2", exportroutes);
```
Comment (me97esn) [inline server/server.js:58]: :+1:

**Review by me97esn — COMMENTED** [review]
This mostly look good, and can be merged.
I have one question though: from looking at the code, it's not obvious what happens if the user doesn't exist in Ladok. It looks like it might cause an error to be thrown, or it might return null/undefined. I guess either of them is fine, but the code covers both scenarios where only one is ever reached. 
So this PR should either remove the null checks, or the try/catch phrase around getStudentData imo

File: server/export.js (hunk 1 above)
Comment (exacs) [reply server/export.js:104]: You are right! They are not needed
